### PR TITLE
Create release_management.md

### LINKED ISSUE #143: Article: Release Management

### PR DESCRIPTION
**Pull request by VinulaUthsara**
Closes #143

## If an article related PR

- [x] I have checked that my suggestion follows the Contribution guide

- [x]  I have added which issue this PR closes

- [x] I have 0 warnings from Danger.js

- [x] If I have added images, they are in /files/ folder

## If other PR

- [] I have added which issue this PR closes

### CODE REVIEW

File: dangerfile.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+"use strict";
```
Comment (prokopsimek) [inline dangerfile.js:1]: @VinulaUthsara please, remove the transpiled code and add it to `.gitignore` 👍

File: package.json [hunk 2]
```diff
@@ -17,5 +17,8 @@
     "prettier": "^1.19.1",
     "ts-node": "^8.5.0",
     "typescript": "^3.7.2"
+  },
+  "dependencies": {
+    "yarn": "^1.22.10"
```
Comment (prokopsimek) [inline package.json:22]: ✅

**Review by prokopsimek — CHANGES_REQUESTED** [review]
@VinulaUthsara Thank you for your pull request! Just please remove the javascript file and add it to `.gitignore`.

@ridlees Do you have any notes to the article?

File: dangerfile.js (hunk 1 above)
Comment (VinulaUthsara) [reply dangerfile.js]: dangerfile.js file removed, but causes a "deploy to preview" check to fail.

**Comment by prokopsimek** [comment]
@VinulaUthsara Thanks! There is a timeout error after 30 secs. I will have to optimize the preview processing. It's not your fault. 👍

**Review by prokopsimek — APPROVED** [review]

File: practices/release_management.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+- [Poor code quality](/problems/poor-code-quality)
+- [Increased cost](/problems/increased-cost)
+- [Demotivated Team](/problems/demotivated-team)
+- [Disconnect Between Business and IT](/problems/disconnect-between-business-and-it)
+- [Bad product-market fit](/problems/bad-product-market-fit)
+
+## How to Implement Release Management
+
+1. **Plan the release**
+   Planning stage requires a lot of time for structuring the release from start to finish. A strong release plan will help teams stay on track and meet expected standards, aims and requirements.
```
Comment (ridlees) [inline practices/release_management.md:86]:
```suggestion
 The planning stage requires a lot of time for structuring the release from start to finish. A well-thought release plan will help teams stay on track and meet expected standards, aims, and requirements.
```

File: practices/release_management.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+2. **Build the release**
+   This stage involves the actual development (designing and building) of the product according to requirements defined during the release plan. After dealing with any issues that surfaced, the build must be subjected to real-world scenario testing. As the development team builds the product, it is tested for user acceptance by using a testing environment. This helps the team find bugs or issues that can occur in the real world. If any issues are identified, the build is sent back for development. The work flow becomes cyclic as the build is sent from the 'development' stage to 'testing' stage and vice versa until the release is approved.
```
Comment (ridlees) [inline practices/release_management.md:89]:
```suggestion
This stage involves the actual development (designing and building) of the product according to requirements defined during the release plan. After dealing with any issues that surfaced, the build must be subjected to real-world scenario testing. As the development team builds the product, it is tested for user acceptance by using a testing environment. Testing helps the team find bugs or issues that can occur in the real world. As the testing team identifies errors they send the build back for development. The workflow becomes cyclic after the build moves the 'development' stage to the 'testing' stage and vice versa until the release is approved.
```

File: practices/release_management.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+3. **User acceptance testing**
+   This involves the gathering of feedback from end users by allowing them to use an early version (free/beta/trial) of the product. This version is shared either online or with a large group of employees of a company. This is a vital step for the release management due to the large quantity of data gathered and fixes needed in order to prepare the product for its official launch.
```
Comment (ridlees) [inline practices/release_management.md:92]:
```suggestion
User Acceptance testing involves the gathering of feedback from end-users by allowing them to use an early version (alpha/beta/trial) of the product. Trials are shared either online or with a large group of employees of a company. Sharing is a vital step for the release management due to the large quantity of data gathered and fixes needed to prepare the product for its official launch.
```

File: practices/release_management.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+4. **Prepare the release**
+   This stage involves the final changes done to the product after considering everything learned from User Acceptance Testing phase. Release preparation also includes a final quality report by the Quality Assurance (QA) team. During this time the QA team will conduct final checks to see if the build meets the minimum standards of acceptability and bussiness requirements which was defined in the release plan. Once this is completed, the functional team will validate findings and work on finalizing the release for deployment. The approval of the product manager is needed before deploying the build into a live environment.
```
Comment (ridlees) [inline practices/release_management.md:95]:
```suggestion
This stage involves the final changes done to the product after considering everything learned from the User Acceptance Testing phase. Release preparation also includes a final quality report by the Quality Assurance (QA) team. During this time, the QA team will conduct their checks to see if the build meets the minimum standards of acceptability and business requirements defined in the release plan. Afterward, the functional team will validate findings and work on finalizing the release for deployment. Before deploying the build into a live environment, the product manager should approve the changes.
```

File: practices/release_management.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+5. **Deploy the release**
+   This is the final stage where the product is released to a live production environment. Apart from simply pushing out a build into production, this stage requires the need to educate the end users and your company. This can be achieved with the messaging, articles, documentation or demonstrations. This helps users to stay notified and updated on how to operate the new features of the release productively and efficiently. Finally, the development team should asess the performance of the release and discuss how to identify and document and issues to be addressed in the next iteration.
```
Comment (ridlees) [inline practices/release_management.md:98]:
```suggestion
Deployment is the final stage where the product release to a live production environment happens. Apart from pushing out a build into production, this stage requires the need to educate the end-users and your company. You can help it with the messaging, articles,  documentation, or demonstrations. Users should be notified and updated on how to operate the new features of the release productively and efficiently. Finally, the development team should assess the performance of the deployment and discuss how to identify and document and issues to be addressed in the next iteration.
```

File: practices/release_management.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+## Common Pitfalls of Release Management
+
+- **Some strategies work better than others:**
+  To have a better outcome, pitfalls of a product release must be examined and a more manageable release should be considered.
+
+- **Coordinating the release:**
+  When designing the product, special thought must be given to its introduction, target user group, support offered and frequency of updates.
```
Comment (ridlees) [inline practices/release_management.md:106]:
```suggestion
 When designing the product, give a thought to its introduction, target user group, the support offered, and frequency of updates.
```

File: practices/release_management.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+- **Conflict of user groups:**
+  While one group prefers new features and updates the other may not. A strategy is needed to handle both groups and proceed with releases when needed.
+
+- **Downtime for testing:**
+  When the release is tested either automatically, manually or both the product is not to be changed unless it is a bug fix.
```
Comment (ridlees) [inline practices/release_management.md:112]:
```suggestion
The product release cycle waits on bug fixes.
```

File: practices/release_management.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+- **Branching and Versioning:**
```
Comment (ridlees) [inline practices/release_management.md:114]:
```suggestion
- Lack of branch and version control.
```

File: practices/release_management.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+5. **Deploy the release**
+   This is the final stage where the product is released to a live production environment. Apart from simply pushing out a build into production, this stage requires the need to educate the end users and your company. This can be achieved with the messaging, articles, documentation or demonstrations. This helps users to stay notified and updated on how to operate the new features of the release productively and efficiently. Finally, the development team should asess the performance of the release and discuss how to identify and document and issues to be addressed in the next iteration.
+
+## Common Pitfalls of Release Management
+
```
Comment (ridlees) [inline practices/release_management.md:101]: We don't usually explain the pitfalls in our articles. Do you think you could rewrite them into single bullet point sentences?

**Review by ridlees — CHANGES_REQUESTED** [review]
Hello @VinulaUthsara, Thank you for the great content. I liked reading through it, it's well structured and well written. I have a few suggestions, please, give them a look.

File: practices/release_management.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+- Productivity is improved by establishing standard release tools and processes.
+- Providing ample time for impact analysis by Change-Advisory Board (CAB).
+- Documenting and establishing of auditable release controls is ensured.
+- Communication of releases to the development team, client representatives, clients, and other organizations.
+- Procedures are streamlined in order to achieve a balance between complexity and required controls of the release.
+- Lessons learned from the release management process you can apply to other areas.
+
+## Problems Release Management Solves
+
+- Unsuccessful product
```
Comment (ridlees) [inline practices/release_management.md:75]:
```suggestion
- [Unsuccessful product](/problems/unhappy-clients)
```

```

File: practices/release_management.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+- coordination
+- branching
+- versioning
+---
+
+# Release Management
+
+**TL;DR**
+
+In the world of business and IT, software systems, resources and development processes  expand in distribution and complexity. How doas Release Management help this? Read more in this article.
```
Comment (ridlees) [inline practices/release_management.md:42]:
```suggestion
In the world of business and IT, software systems, resources, and development processes expand in distribution and complexity. How does Release Management help with controlling your deployments?

File: practices/release_management.md [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+  While one group prefers new features and updates the other may not. A strategy is needed to handle both groups and proceed with releases when needed.
+
+- **Downtime for testing:**
+The product release cycle waits on bug fixes.
+
+- Lack of branch and version control.
+  A versioning system with multiple branching support helps managing features of the product. One branch maybe volatile due to new features being introduced while the other maybe stable.
+
+## Resources for Release Management
+
```
Comment (ridlees) [inline practices/release_management.md:119]:
```suggestion

- [Release Management Definition](https://en.wikipedia.org/wiki/Release_management)
```

**Review by ridlees — APPROVED** [review]

**Pull request merged**